torch/csrc/autograd/variable.h

```diff
@@ -333,40 +303,82 @@ struct Variable::Impl : public at::TensorImpl {
 
   std::shared_ptr<Function> get_grad_accumulator();
   virtual std::shared_ptr<Function>& get_grad_fn() {
-    return grad_fn;
+    return grad_fn_;
+  }
+
+  virtual const Variable& base() const {
+    throw std::runtime_error("Can't get base of non-view Variable");
+  }
+
+  /// Sets the `requires_grad` property of `Variable`. This should be true for
+  /// leaf variables that want to accumulate gradients, and false for all other
+  /// variables.
+  void set_requires_grad(bool requires_grad) noexcept override {
```

Comment:
Can you please mark inherited virtual functions as virtual?

Reply:
I know it's redundant but it's incredibly important. I never understand why people don't propagate this keyword, because it completely changes the semantics and perf characteristic of a function.

Reply:
Ohh I see now. Never mind then, that's good!